compiler/rustc_mir_transform/src/const_prop_lint.rs

```diff
@@ -501,12 +501,23 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 return None;
             }
 
+            Rvalue::Len(len_place) => {
```

Comment:
I think we should make the same change in the `const_prop` pass as well:

https://github.com/rust-lang/rust/blob/9bbbf60b0442f1d56fc39f30274be77acc79164c/compiler/rustc_mir_transform/src/const_prop.rs#L624

but that doesn't have to be in this PR 🙂

Reply:
afaicr this is not necessary, because there are optimizations taking care of this before const prop runs